File: p2pusenet/extra/scripting-examples/DetailedExample/example2.rb
#!/usr/bin/env ruby
#

#Note: I say this is ruby, but it's truthfully JRuby,
#You can change the path env to JRuby if you desire.
#

print "Starting the second script\n"

require "java"
print "trying to include all of java.\n"
print "This will cause name conflicts with things like 'String'\n"
print "\t...so we will put it in a module called 'Java'\n"
module Java
	include_package 'java'
end

include_class 'javax.script.ScriptEngineManager'
include_class 'javax.script.ScriptEngine'

print "Trying to get a Script Engine Manager...\n"
mgr = ScriptEngineManager.new

print "Trying to get a Python engine... in ruby\n"
pyEng = mgr.getEngineByName("python")

print "Evaluating python...\n"
pyEng.eval("print 'This is python inside a ruby script!'")

print "SUCCESS!\n"

File: p2pusenet/extra/scripting-examples/DetailedExample/example2.py
#!/usr/bin/env python

# I say this is python, but it's truly jython.
# We're going to extensively call java code in here

print "Start of the example2.py script"
#from java import Object
#from java import System
#let's try this:
print "trying to import everything from java"
from java import *
from javax.script import *

print "Trying to create a script engine manager"
mgr = ScriptEngineManager()

print "Trying to get a ruby engine...  in python"
rbEng = mgr.getEngineByName("ruby");

print "Evaluating ruby..."
rbEng.eval("print 'Hello from ruby, inside a python script\n'")

print "Setting our variables to None"
mgr, rbEng = None, None

print "SUCCESS!!!!"


